homeassistant/components/media_player/ps4.py

```diff
@@ -0,0 +1,341 @@
+            self.load_games()
+
+    def save_games(self, games):
+        """Save games to file."""
+        file = self._games_filename
+        try:
+            save_json(file, games)
+        except OSError as error:
+            _LOGGER.error("Could not save game list, %s", error)
+
+    def add_games(self, titleid, app_name):
+        """Add games to list."""
+        games = self.load_games()
+        if titleid is not None and titleid not in games:
+            game = {titleid: app_name}
+            games.update(game)
+            self.save_games(games)
+
+    @property
+    def entity_picture(self):
+        """Return picture."""
+        if self._state == STATE_OFF:
+            return None
+
+        image_hash = self.media_image_hash
+        if image_hash is not None:
+            return ENTITY_IMAGE_URL.format(
+                self.entity_id, self.access_token, image_hash)
+
+        if self._media_content_id is None:
+            return None
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._state
+
+    @property
+    def icon(self):
+        """Icon."""
+        return ICON
+
+    @property
+    def media_content_id(self):
+        """Content ID of current playing media."""
+        return self._media_content_id
+
+    @property
+    def media_content_type(self):
+        """Content type of current playing media."""
+        return MEDIA_TYPE_MUSIC
+
+    @property
+    def media_image_url(self):
+        """Image url of current playing media."""
+        if self._media_content_id is None:
+            return MEDIA_IMAGE_DEFAULT
+        try:
+            return self._media_image
+            # return self._gamesmap[self._media_content_id]
+        except KeyError:
+            return MEDIA_IMAGE_DEFAULT
+
+    @property
+    def media_title(self):
+        """Title of current playing media."""
+        return self._media_title
+
+    @property
+    def supported_features(self):
+        """Media player features that are supported."""
+        return SUPPORT_PS4
+
+    @property
+    def source(self):
+        """Return the current input source."""
+        return self._source
+
+    @property
+    def source_list(self):
+        """List of available input sources."""
+        return self._source_list
+
+    def turn_off(self):
+        """Turn off media player."""
+        self._ps4.standby()
+
+    def turn_on(self):
+        """Turn on the media player."""
+        self._ps4.wakeup()
+
+    def media_pause(self):
+        """Send keypress ps to return to menu."""
+        self._ps4.remote_control('ps')
+
+    def media_stop(self):
+        """Send keypress ps to return to menu."""
+        self._ps4.remote_control('ps')
+
+    def select_source(self, source):
+        """Select input source."""
+        if self._source_selected is not None:
```

Comment:
So one can only select source one time? The only time it resets is if you manually turn off the game, is there any way around this?

Reply:
As of now. The PS4 prompts you to close current application. In testing I have found that you can switch sources multiple times.

Reply:
When source switches completely, the state should change to idle before booting game. Have you tried testing it?

Reply:
Tested and it works nicely